src/modules/Threshold/Module.js

```diff
@@ -25,7 +31,13 @@ module.exports = function ImageThreshold(options) {
     }).then(function (imageData) {
       image = new Image();
       image.onload = function onLoad() {
-        if (options.output) options.output(image);
+        this.output = {
```

Comment:
`this` is being used in the wrong context; The output needs to be set for the "step" object. Like this:

```js
function draw (inputImage) {
  var step = this;
  ...
  ...
  ...
      image = new Image();
      image.onload = function onLoad() {
        step.output = {
                                  format: input.type, 
                                  src: image.src
                               }
       options.step.output = step.output.src;
       callback();
       UI.onComplete(options.step);
}
```

Reply:
Ah, let's refactor existing modules to save `this` into `var step` to avoid this ambiguity.

Reply:
Ah, i see that existing modules are already this way! Great, it must just be that the Threshold module wasn't.